File: src/pysrc/test/int/test_data_client.py
import pytest
from pysrc.data_client import DataClient


def test_integration() -> None:
    data_client = DataClient()

    result = data_client.get_data()

    for trade in result:
        assert trade["type"]
        assert trade["amount"]
        assert trade["timestampms"]
        assert trade["price"]
        assert (
            trade["type"] == "trade"
            or trade["type"] == "buy"
            or trade["type"] == "sell"
        )
        assert type(float(trade["amount"])) is float
        assert type(trade["timestampms"]) is int
        assert type(float(trade["price"])) is float

File: src/cppsrc/test/test_data_client.cpp
#include "cppsrc/data_client.hpp"
#include <gtest/gtest.h>

TEST(DataClientTests, ParseMessageTest)
{
    DataClient client = DataClient();
    nlohmann::json result = client.get_data();

    for (int i = 0; i < result.size(); i++) {
        EXPECT_TRUE(result[i]["type"] == "sell" || result[i]["type"] == "buy");
        EXPECT_TRUE(result[i]["amount"] > 0);
        EXPECT_TRUE(result[i]["price"] > 0);
    }
}
